plugins/policy/processor/processor.go

```diff
@@ -462,3 +512,71 @@ func (pp *PolicyProcessor) getPoliciesAssignedToPod(pod *podmodel.Pod) (policies
 	}
 	return policies
 }
+
+// getPoliciesAssignedToNamespace returns all policies currently assigned to a namespace.
+func (pp *PolicyProcessor) getPoliciesAssignedToNamespace(ns *nsmodel.Namespace) (policies map[policymodel.ID]*policymodel.Policy) {
+	policies = make(map[policymodel.ID]*policymodel.Policy)
+	labelsMap := make(map[string]string)
+
+	nsLabels := ns.Label
+	for _, nsLabel := range nsLabels {
+		labelsMap[nsLabel.Key] = nsLabel.Value
+	}
+
+	// Fetch data of all policies from the cache.
+	allPolicies := pp.Cache.ListAllPolicies()
+	dataPolicies := []*policymodel.Policy{}
+	for _, policy := range allPolicies {
+		found, policyData := pp.Cache.LookupPolicy(policy)
+
+		if !found {
+			continue
+		}
+		dataPolicies = append(dataPolicies, policyData)
+	}
+
+	// Select policies that match pod's labels.
+	for _, dataPolicy := range dataPolicies {
+		dataPolicyID := policymodel.GetID(dataPolicy)
+		if len(dataPolicy.IngressRule) == 0 {
+			continue
+		} else {
+			for _, ingressRules := range dataPolicy.IngressRule {
+				for _, ingressRule := range ingressRules.From {
+					matchLabels := []*policymodel.Policy_Label{}
+
+					if ingressRule.Namespaces != nil {
+						matchLabels = ingressRule.Namespaces.MatchLabel
+						for _, label := range matchLabels {
+							if labelsMap[label.Key] != label.Value {
+								break
```

Comment:
looks like you added `policies[dataPolicyID] = dataPolicy` anyway, why need this for loop at all ?

Reply:
Yup, policies[dataPolicyID] = dataPolicy should be in the for loop, fixed